modules/Classroom/components/ConfirmDeleteExamModal/ConfirmDeleteExamModal.tsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from "react";
+
+import { Modal, Button, Text, Loader, Group } from "@mantine/core";
+import { showNotification } from "@mantine/notifications";
+
+import { useDeleteExamMutation } from "@/shared/redux/rtk-apis/exams/exams.api";
+
+import { IConfirmDeleteExamModalProps } from "./ConfirmDeleteExamModal.interface";
+
+const ConfirmDeleteExamModal: React.FC<IConfirmDeleteExamModalProps> = ({
+  examId,
+  isOpen,
+  onClose,
+}) => {
+  const [deleteExam, { isLoading }] = useDeleteExamMutation();
+  const [error, setError] = useState<string | null>(null);
+
+  const handleDelete = async () => {
+    try {
+      await deleteExam({ examId: examId }).unwrap();
+      showNotification({
+        title: "Success",
+        message: "Exam deleted successfully",
+        color: "blue",
+      });
+      onClose();
+    } catch (err) {
+      setError("Failed to delete exam");
+      showNotification({
+        title: "Error",
+        message: "Could not delete the exam. Please try again.",
+        color: "red",
+      });
+    }
+  };
+
+  return (
+    <Modal opened={isOpen} onClose={onClose} title="Confirm Delete Exam" centered>
+      {isLoading ? (
+        <Loader />
```

Comment:
I believe it was discussed that loading spinner would be displayed inside the submit/delete button while loading